Sources/IBLinterFrontend/Commands/ValidateCommand.swift

```diff
@@ -8,75 +8,50 @@
 import Foundation
 import IBDecodable
 import IBLinterKit
-import Commandant
+import ArgumentParser
 
-struct ValidateCommand: CommandProtocol {
-    typealias Options = ValidateOptions
-    typealias ClientError = Options.ClientError
+struct ValidateCommand: ParsableCommand {
+    static let configuration = CommandConfiguration(commandName: "lint", abstract: "Print lint warnings and errors")
 
-    let verb: String = "lint"
-    var function: String = "Print lint warnings and errors (default command)"
+    @Option(name: .long, help: "validate project root directory", completion: .directory)
+    var path: String?
+    @Option(name: .long, help: "the reporter used to log errors and warnings")
+    var reporter: String?
+    @Option(name: .long, help: "custom IBLinterfile.swift", completion: .file(extensions: ["swift"]))
+    var iblinterFilePath: String?
+    @Option(name: .long, help: "the path to IBLint's configuration file", completion: .file())
+    var configurationFile: String?
+    @Argument(help: "included files/paths to lint. This is ignored if you specified included paths in your yml configuration file.",
+              completion: .file())
+    var included: [String] = []
 
-    func run(_ options: ValidateCommand.Options) -> Result<(), ValidateCommand.ClientError> {
-        let workDirectoryString = options.path ?? FileManager.default.currentDirectoryPath
+    func run() throws {
+        let workDirectoryString = path ?? FileManager.default.currentDirectoryPath
         let workDirectory = URL(fileURLWithPath: workDirectoryString)
         guard FileManager.default.isDirectory(workDirectory.path) else {
             fatalError("\(workDirectoryString) is not directory.")
         }
 
-        var config = (try? Config(url: options.deriveConfigurationFile())) ?? Config.default
+        var config = (try? Config(url: deriveConfigurationFile())) ?? Config.default
         if config.disableWhileBuildingForIB &&
             ProcessInfo.processInfo.compiledForInterfaceBuilder {
-            return .success(())
+            return
         }
         if config.included.isEmpty {
-            config.included = options.included
+            config.included = included
         }
         let validator = Validator()
         let violations = validator.validate(workDirectory: workDirectory, config: config)
 
-        let reporter = Reporters.reporter(from: options.reporter ?? config.reporter)
+        let reporter = Reporters.reporter(from: self.reporter ?? config.reporter)
         let report = reporter.generateReport(violations: violations)
         print(report)
 
         let numberOfSeriousViolations = violations.filter { $0.level == .error }.count
         if numberOfSeriousViolations > 0 {
-            exit(2)
-        } else {
-            return .success(())
+            throw ValidationError("")
```

Comment:
Here I used `ValidationError` since it provides "an exit code that indicates that the user provided invalid input." [link](https://github.com/apple/swift-argument-parser/blob/cca8f80939f43cb631f6a1c3e7003ce280831542/Sources/ArgumentParser/Parsable%20Properties/Errors.swift#L68)

but now I feel like it might be better to define custom Error and throw that.

What do you think?

Reply:
How about throwing `ArgumentParser.ExitCode.failure` directly?

Reply:
It makes total sense. Fixed at 1fbe629